jxl/src/bit_reader.rs

```diff
@@ -167,9 +187,11 @@ impl<'a> BitReader<'a> {
             // Prevent the returned bitreader from over-reading.
             ret.data = &ret.data[..n - bytes_in_buf];
         } else {
-            ret.bits_in_buf -= n * 8;
```

Comment:
it should be mentioned in the commit message that this also fixes some bugs

Reply:
Good point...